gcc/rust/hir/tree/rust-hir-expr.h

```diff
@@ -3721,10 +3722,11 @@ struct MatchArm
   // Constructor for match arm with a guard expression
   MatchArm (std::vector<std::unique_ptr<Pattern> > match_arm_patterns,
 	    std::unique_ptr<Expr> guard_expr = nullptr,
-	    AST::AttrVec outer_attrs = AST::AttrVec ())
```

Comment:
I think for HIR we should make these mandatory arguments. So you will need to remove the '=' and update the HIR lowering code appropriately when you get the compile errors will show you where to update it and you should get this from the AST version.